### PR TITLE
Jvenstad/submit

### PR DESCRIPTION
**Pull request by jonmv**
Who's a good reviewer for this, I wonder ... Perhaps @tokle or @andreer ?

### CODE REVIEW

File: tenant-cd/src/main/java/ai/vespa/hosted/cd/ProductionTest.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // Copyright 2018 Yahoo Holdings. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
-package com.yahoo.vespa.tenant.cd;
+package ai.vespa.hosted.cd;
```
Comment (andreer) [inline tenant-cd/src/main/java/ai/vespa/hosted/cd/ProductionTest.java]: I'm surprised! Is this something we should generally be doing in new code?

Comment (jonmv) [reply tenant-cd/src/main/java/ai/vespa/hosted/cd/ProductionTest.java]: That's what I was told :)

**Comment by jonmv** [comment]
PTAL @bjorncs

**Review by bjorncs — COMMENTED** [review]
See comments

File: security-utils/src/test/java/com/yahoo/security/KeyUtilsTest.java [hunk 2]
```diff
@@ -17,6 +17,32 @@
  */
 public class KeyUtilsTest {
 
+    private static final String rsaPemPublicKey = "-----BEGIN PUBLIC KEY-----\n" +
```
[the remark this answers is not in the document]
Comment (jonmv) [reply security-utils/src/test/java/com/yahoo/security/KeyUtilsTest.java]: I think verifying that keys generated by `openssl` are parsed correctly is valuable in itself, so I'll leave the keys there, but I'll provide the commands used to generate them. 

I'll take a look at serialising the public keys, too, like what's done for private keys.

File: hosted-api/src/test/java/ai/vespa/hosted/api/SignaturesTest.java [hunk 3]
```diff
@@ -124,13 +58,9 @@
...
-    @Test
-    public void testEncryption() {
-        Key privateKey = KeyUtils.fromPemEncodedPrivateKey(pemPrivateKey);
-        Key publicKey = KeyUtils.fromPemEncodedPublicKey(pemPublicKey);
-
-        assertArrayEquals(message, decrypted(encrypted(message, privateKey), publicKey));
-        assertArrayEquals(message, decrypted(encrypted(message, publicKey), privateKey));
+    @BeforeClass
+    public static void register() {
+        Security.addProvider(new BouncyCastleProvider());
```
[the remark this answers is not in the document]
Comment (jonmv) [reply hosted-api/src/test/java/ai/vespa/hosted/api/SignaturesTest.java]: Fixed by specifying provider, in code moved to `security-utils`.

File: security-utils/src/test/java/com/yahoo/security/KeyUtilsTest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (jonmv) [reply security-utils/src/test/java/com/yahoo/security/KeyUtilsTest.java]: Comments with `openssl` commands added.

As we discovered, we should try to generate keys with the `EC` algorithm when reading such keys with Bouncy Castle.

**Comment by jonmv** [comment]
Approved, @bjorncs ? :)

**Review by bjorncs — COMMENTED** [review]
👍

**Pull request merged**